src/System.Device.Gpio.Tests/RaspberryPi4DriverTests.Linux.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Device.Gpio.Drivers;
+using System.Threading;
+using Xunit;
+
+namespace System.Device.Gpio.Tests
+{
+    public class RaspberryPi4DriverTests : GpioControllerTestBase
+    {
+        /// <summary>
+        /// Leave this pin open (unconnected) for the tests
+        /// </summary>
+        private const int OpenPin = 23;
+
+        protected override GpioDriver GetTestDriver()
+        {
+            return new RaspberryPi4Driver();
+        }
+
+        protected override PinNumberingScheme GetTestNumberingScheme() => PinNumberingScheme.Logical;
+
+        [Fact]
+        public void InputPullResistorsWork()
+        {
+            // This is a hack to test whether we're running on a Pi4 without duplicating that detection logic. 
+            var bestDriver = GpioController.GetBestDriverForBoard();
+            if (!(bestDriver is RaspberryPi4Driver))
+            {
+                bestDriver.Dispose();
+                return;
+            }
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), bestDriver))
+            {
+                controller.OpenPin(OpenPin, PinMode.InputPullUp);
+                Thread.Sleep(10);
```

Comment:
Can you please explain why you are pausing for 10ms?

Reply:
Not really. I was first expecting it to take a moment to settle, but (as the remainder of the test shows), the change happens instantly. So this is not needed.